src/constructors/circulant.c

```diff
@@ -0,0 +1,97 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_constructors.h"
+#include "igraph_interface.h"
+
+/**
+ * \function igraph_circulant
+ * \brief Creates a circulant graph.
+ *
+ * A circulant graph <code>G(n, l)</code> consists of \p n vertices \c v_0, ...,
+ * \c v_(n-1) such that for each \c l_i in the list of offsets \p l, \c v_j is
+ * connected to <code> v_(j + l_i mod n) </code> for all j.
```

Comment:
I think this should be `v_((j + l_i) mod n)` just to be clear that the modulo applies to the entire sum. I'm not sure what the "official" precedence of the modulo operator is.

Reply:
Done.